version: '3'

services:
  db:
    image: postgres:10
    container_name: auth_db_postgres
    environment:
      POSTGRES_DB: Fausto
      POSTGRES_USER: root
      POSTGRES_PASSWORD: faustoauthdb24$$ # solo con un dolar $
    ports:
      - "5464:5432"
    volumes:
      - ./docker/postgres_db/initdb/:/docker-entrypoint-initdb.d
      - ./data_temp/pgdb/backup:/pgdb/backup
      - ./data_temp/pgdb/script:/pgdb/script
  db_admin:
    build:
      context: .
      dockerfile: ./docker/pgadmin/Dockerfile
    container_name: auth_db_admin_postgres
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@fausto.app
      PGADMIN_DEFAULT_PASSWORD: admin24$$ # solo con un dolar $
    ports:
      - "8081:80"
    volumes:
      - ./data_temp/pgadmin:/var/lib/pgadmin
      - ./docker/pgadmin/data.json:/pgadmin4/servers.json
    logging:
      driver: none
  # api:
  #   container_name: auth_api
  #   build: 
  #     context: .
  #     dockerfile: ./docker/api/api.Dockerfile
  #   volumes:
  #     - ./api/src:/api/src
  #     - ./api/src/.env:/api/src/.env
  #   command: >    
  #     sh -c "more .env && python run.py"
  #   restart: always
  #   ports:
  #     - "5111:5000"
  fastapi:
    container_name: auth_fastapi
    build: 
      context: .
      dockerfile: ./docker/fastapi/fastapi.Dockerfile
    volumes:
      - ./fastapi/:/fastapi/
      - ./fastapi/.env:/fastapi/.env
    command: uvicorn run:app --reload  --host 0.0.0.0 --port 5000  --log-level 'debug'
    restart: always
    ports:
      - "9024:5000"
    depends_on:
      - db
      - db_admin
      - redis
  redis:
    image: redis
    container_name: auth_cache
    expose:
      - 6379

